pythonFiles/vscode_pytest/__init__.py

```diff
@@ -0,0 +1,227 @@
+    # TODO: add error checking.
+
+
+def build_test_tree(session) -> Tuple[Union[TestNode, None], List[str]]:
+    # Builds a tree of tests from the pytest session.
+    errors: List[str] = []
+    session_node: TestNode = create_session_node(session)
+    session_children_dict: dict[str, TestNode] = {}
+    file_nodes_dict: dict[pytest.Module, TestNode] = {}
+    class_nodes_dict: dict[str, TestNode] = {}
+
+    for test_case in session.items:
+        test_node: TestItem = create_test_node(test_case)
+        if type(test_case.parent) == DoctestTextfile:
+            try:
+                parent_test_case: TestNode = file_nodes_dict[test_case.parent]
+            except KeyError:
+                parent_test_case: TestNode = create_doc_file_node(test_case.parent)
+                file_nodes_dict[test_case.parent] = parent_test_case
+            parent_test_case["children"].append(test_node)
+        elif type(test_case.parent) is pytest.Module:
+            try:
+                parent_test_case: TestNode = file_nodes_dict[test_case.parent]
+            except KeyError:
+                parent_test_case: TestNode = create_file_node(test_case.parent)
+                file_nodes_dict[test_case.parent] = parent_test_case
+            parent_test_case["children"].append(test_node)
+        else:  # should be a pytest.Class
+            try:
+                test_class_node: TestNode = class_nodes_dict[test_case.parent.name]
+            except KeyError:
+                test_class_node: TestNode = create_class_node(test_case.parent)
+                class_nodes_dict[test_case.parent.name] = test_class_node
+            test_class_node["children"].append(test_node)
+            parent_module: pytest.Module = test_case.parent.parent
+            # Create a file node that has the class as a child.
+            try:
+                test_file_node: TestNode = file_nodes_dict[parent_module]
+            except KeyError:
+                test_file_node: TestNode = create_file_node(parent_module)
+                file_nodes_dict[parent_module] = test_file_node
+            test_file_node["children"].append(test_class_node)
+            # Check if the class is already a child of the file node.
+            if test_class_node not in test_file_node["children"]:
+                test_file_node["children"].append(test_class_node)
+    created_files_folders_dict: dict[str, TestNode] = {}
+    for file_module, file_node in file_nodes_dict.items():
+        # Iterate through all the files that exist and construct them into nested folders.
+        root_folder_node: TestNode = build_nested_folders(
+            file_module, file_node, created_files_folders_dict, session
+        )
+        # The final folder we get to is the highest folder in the path and therefore we add this as a child to the session.
+        if root_folder_node.get("id_") not in session_children_dict:
+            session_children_dict[root_folder_node.get("id_")] = root_folder_node
```

Comment:
```suggestion
        # The final folder we get to is the highest folder in the path and therefore we add this as a child to the session.
        root_folder_node_id = root_folder_node.get("id_")
        if root_folder_node_id is not None and root_folder_node_id not in session_children_dict:
            session_children_dict[root_folder_node_id] = root_folder_node
```

If `"id_"` is always expected to be there as a key, there's no need to protect the access behind `dict.get()`.